app/Filament/Resources/MemberResource.php

```diff
@@ -0,0 +1,268 @@
+                            ->label('Mark as inactive')
+                            ->requiresConfirmation()
+                            ->action(fn(Member $record) => tap($record, function ($record) {
+                                $record->update(['status' => 'inactive']);
+                                Notification::make()
+                                    ->title('Member has been deactivated')
+                                    ->danger()
+                                    ->send();
+                            }))
+                            ->visible(fn($record) => $record->status === 'active'),
+                    ])->dropdown(false),
+                    Tables\Actions\ActionGroup::make([
+                        Tables\Actions\Action::make('heading_actions')
+                            ->label('Record Actions')
+                            ->disabled()
+                            ->color('gray'),
+                        Tables\Actions\ViewAction::make(),
+                        Tables\Actions\EditAction::make()->hiddenLabel(),
+                        Tables\Actions\DeleteAction::make()->hiddenLabel(),
+                    ])->dropdown(false),
+                ]),
+            ])->recordUrl(fn($record): string => route('filament.admin.resources.members.view', $record->id))
+            ->bulkActions([
+                Tables\Actions\BulkActionGroup::make([
+                    Tables\Actions\DeleteBulkAction::make(),
+                    Tables\Actions\ForceDeleteBulkAction::make(),
+                    Tables\Actions\RestoreBulkAction::make(),
+                ]),
+            ]);
+    }
+
+    public static function getRelations(): array
+    {
+        return [
+            //
+        ];
+    }
+
+    /**
+     * Add infolist to the resource.
+     */
+    public static function infolist(Infolist $infolist): Infolist
+    {
+        return $infolist
+            ->schema([
+                Section::make()
+                    ->heading(function (Member $record): HtmlString {
+                        $variant = $record->status === 'active' ? 'primary' : 'danger';
+
+                        $badge = '<span '
+                            . 'style=" margin-left: 0.5rem; '
+                            . '--c-50:var(--' . $variant . '-50);'
+                            . '--c-400:var(--' . $variant . '-400);'
+                            . '--c-600:var(--' . $variant . '-600);" '
+                            . 'class="inline-flex fi-badge items-center justify-center gap-x-1 '
+                            . 'rounded-md text-xs font-medium ring-1 ring-inset px-2 py-1 '
+                            . 'fi-color-custom bg-custom-50 text-custom-600 ring-custom-600/10 '
+                            . 'dark:bg-custom-400/10 dark:text-custom-400 dark:ring-custom-400/30 '
+                            . 'fi-color-' . $variant . '">'
+                            . ucfirst($record->status)
+                            . '</span>';
```

Comment:
this can simplified by using filament blade component for badge

https://filamentphp.com/docs/3.x/support/blade-components/badge#changing-the-color-of-the-badge